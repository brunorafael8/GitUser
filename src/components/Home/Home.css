.form__search {
  width: 50%;
  padding: 1rem 0;
  justify-content: space-around;
}


.form__search input {
  flex-grow: 2;
  margin: 10px;
}

.form__search button,
.form__search input[type="submit"], {
  flex-grow: 1;

}
