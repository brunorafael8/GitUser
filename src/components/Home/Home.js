'use strict'

import React, { PureComponent } from 'react'
import { QueryRenderer, graphql } from 'react-relay'

import styled from 'styled-components';

import environment from 'src/environment'

import User from 'components/User/User'

import Header from 'components/Header/Header'
import Title from 'components/Title/Title';
import Search  from 'components/Search/Search';
import SearchButton from 'components/SearchButton/SearchButton'

import './Home.css';

const query = graphql`
  query HomeQuery ($login: String!) {
    user (login: $login) {
      ... User_user
    }
  }
`;

export default class Home extends PureComponent {
  state = {
    user: null
  };

  onsearch(event) {
    event.preventDefault();
    const user = event.target.querySelector('input[type="text"]').value;
    this.setState({ user });
  }

  render () {
    const { user } = this.state;
    const variables = { login: user };
    return (
      <div>
        <Header>
          <Title>Search for your github user</Title>
          <form className="form form__search" onSubmit={this.onsearch.bind(this)}>
            <Search name="search" placeholder="Ex.: brunorafael8" type="text" />
            <SearchButton value="Buscar">
              Search
            </SearchButton>
          </form>
        </Header>
        {user &&
          <QueryRenderer
            environment={environment}
            query={query}
            variables={variables}
            render={({ error, props }) => {
              if (error) console.error(new Error(error));

              if (props) {
                return <User user={props.user} />
              } else {
                return <div>Searching for {user}, loading ...</div>
              }
            }}
          />
      }
      </div>
    )
  }
}
