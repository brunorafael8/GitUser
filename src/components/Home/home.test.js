'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import Home from './Home'

describe('Test UI', () => {
  it('has the expected UI output', () => {
    const tree = renderer.create(<Home />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
