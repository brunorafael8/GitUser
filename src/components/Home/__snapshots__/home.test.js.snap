// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI has the expected UI output 1`] = `
<div>
  <div
    className="sc-gzVnrw fMOJSM"
  >
    <h1
      className="sc-htoDjs jbJRmU"
    >
      Search for your github user
    </h1>
    <form
      className="form form__search"
      onSubmit={[Function]}
    >
      <input
        className="sc-dnqmqq jrpzfl"
        name="search"
        placeholder="Ex.: brunorafael8"
        type="text"
      />
      <button
        className="sc-iwsKbI xLoYR"
        value="Buscar"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
